src/NuGet.Core/NuGet.Commands/RestoreCommand/CompatibilityChecker.cs

```diff
@@ -324,10 +417,11 @@ internal static bool HasCompatibleAssets(LockFileTargetLibrary targetLibrary)
                 targetLibrary.ContentFiles.Count > 0 ||                               // Shared content
                 targetLibrary.ResourceAssemblies.Count > 0 ||                         // Resources (satellite package)
                 targetLibrary.Build.Count > 0 ||                                      // Build
-                targetLibrary.BuildMultiTargeting.Count > 0;                          // Cross targeting build
+                targetLibrary.BuildMultiTargeting.Count > 0 ||                        // Cross targeting build
+                targetLibrary.ToolsAssemblies.Count > 0;                              // Tools assemblies - This makes a package not backwards compatible potentially, but do tools assets matter
```

Comment:
This should be separate check, a regular project should not see tool packages as compatible. I am guessing that this wouldn't be populated if it wasn't a tools project, but it would be more clear if there were a different method to check tool assets.